---
title: "MED263 R Tutorial"
author: "Jihoon Kim (j5kim@ucsd.edu)"
date: "2/2/2017"
output: html_document
fontsize: 10pt
---

### R  

R is 'GNU S', a freely available language and environment for statistical computing and graphics which provides a wide variety of statistical and graphical techniques

- linear and nonlinear modeling
- statistial tests
- time series analysis
- classification
- clustering
- genomics


Download and install R (precompiled binary distribution) from UCLA <http://cran.stat.ucla.edu>, the nearest R mirror site <http://cran.r-project.org/mirrors.html> 


### R Studio 

RStudio is a set of integrated development environment (IDE) designed to help you be more prudictive with R. It includes 

- console
- syntax-highlighting editor that supports direct code execution
- plotting
- history
- debugging 
- workspace management. 

Download and install RStudio from <https://www.rstudio.com/products/rstudio/download>


### Calculate some numbers

```{r, echo = TRUE }
1 + 2
2^4
log(16, 2)
log(1000, 3)
choose(5, 2)
```


### Data types 

##### 1. Logical
```{r, echo = TRUE }
relapse <- TRUE
typeof( relapse )
relapse
sbp <- c(160, 110, 105, 150, 170)
sbp > 120
diag <- ifelse( sbp > 120, "hypertension", "normal")
```


##### 2. Number
```{r, echo = TRUE }
ldl <- 100
typeof( ldl )
hdl <- 60
vldl <- 25
totalChol <- hdl + ldl + vldl
totalChol 
```


##### 3. Vector of numbers
```{r, echo = TRUE }
cholesterol <- c(177, 193, 195, 209, 226)
typeof( cholesterol )
length( cholesterol )
mean( cholesterol )
max( cholesterol )
```

##### 4. Vector of characters
```{r, echo = TRUE }
genotype <- c("AA", "AG", "GG")
typeof( genotype )
```

##### 5. Factors
```{r, echo = TRUE }
alleles <- c( "AG", "GG", "AA", "AA", "AA", "AA", "AA", "AG", "AG", "GG", "GG", "GG", "GG", "AA")
typeof( alleles )
alleles 
alleles <- factor( alleles )
levels( alleles )
typeof( alleles )
alleles
table( alleles )
```



##### 6. Data Frames 
```{r, echo = TRUE } 
genotype <- c("AA", "AA", "AA",  "AA", "AA", "GG", "GG", "GG", "GG")
mrna <- c(100, 90, 105, 87, 92, 20, 24, 35, 27)
ethnicity <- c("AFR", "AFR", "AFR", "AMR",  "AFR", "ASN", "EUR", "EUR", "EUR")
relapse <- c("yes", "yes", "yes", "no", "yes", "no", "no", "no", "no")
study <- data.frame( genotype, mrna, ethnicity, relapse)
dim( study )
head( study )
```


### Load data from a file 
```{r, echo = TRUE}
# Features are computed from a digitized image of a fine needle aspirate (FNA) of a breast mass. 
# They describe characteristics of the cell nuclei present in the image. 
url <-"http://archive.ics.uci.edu/ml/machine-learning-databases/pima-indians-diabetes/pima-indians-diabetes.data"
pima <- read.table( url, sep=",")
colnames( pima ) <- c("pregnancy", "gtt", "dbp", "skinthickness", "insulin", "bmi", "familyhistory", "age", "diabetes")
head( pima )
dim( pima )
table( pima$diabetes )
```

### Box plot
```{r, echo = TRUE}
boxplot( gtt ~ diabetes, 
         data = pima)
```

### Box plot with added features
```{r, echo = TRUE}
boxplot( gtt ~ diabetes, 
         data = pima, 
         boxwex = 0.3,
         main = "Pima Indians Diabetes Study",
         ylab = "Glucose concentration", 
         xlab = "Diabetes")
```


### Two group comparison, glucose concentration in GTT
```{r, echo = TRUE}
case <- pima$gtt[ pima$diabetes == 1]
control <- pima$gtt[ pima$diabetes == 0]
t.test( case, control ) 
```

### Two group comparison, Diastolic blood pressure (DBP)
```{r, echo = TRUE}
case <- pima$dbp[ pima$diabetes == 1]
control <- pima$dbp[ pima$diabetes == 0]
t.test( case, control ) 
```

### Box plot of DBP
```{r, echo = TRUE}
boxplot( dbp ~ diabetes, 
         data = pima, 
         boxwex = 0.3,
         main = "Pima Indians Diabetes Study",
         ylab = "Diastolic blood pressure", 
         xlab = "Diabetes")
```

### Logistic regression with a binary outcome
```{r, echo = TRUE}
m1 <- glm( diabetes ~ pregnancy + gtt + dbp + bmi + familyhistory, 
           data=pima, 
           family = "binomial")
round( summary( m1 )$coef, 3) 
```

### Forest plot 

#### randomised trials of corticosteroid therapy in premature labour and its effect on neonatal death.
```{r, echo = TRUE}
library( rmeta )

data( cochrane )

print( cochrane )

steroid <- meta.MH(n.trt, n.ctrl, 
                   ev.trt, ev.ctrl,
                   names=name, data=cochrane)
tabletext <- cbind( 
  c("","Study",steroid$names,NA,"Summary"),
  c("Deaths","(steroid)",cochrane$ev.trt,NA,NA),
  c("Deaths","(placebo)", cochrane$ev.ctrl, NA,NA),
  c("","OR",format(exp(steroid$logOR),digits=2),NA,format(exp(steroid$logMH),digits=2))
)

      
m <-     c(NA, NA, steroid$logOR,   NA, steroid$logMH)
l <- m - c(NA, NA, steroid$selogOR, NA, steroid$selogMH)*2
u <- m + c(NA, NA, steroid$selogOR, NA, steroid$selogMH)*2

forestplot(tabletext, 
           m, l, u, zero=0, 
           is.summary=c(TRUE,TRUE,rep(FALSE,8),TRUE),
           clip=c(log(0.1),log(2.5)), 
           xlog=TRUE,
           col=meta.colors(box="royalblue", 
                           line="darkblue", 
                           summary="royalblue"))
```

### Population Stratification using PCA
```{r, echo = TRUE}

# INSTALL REQUIRED R PACKAGES
source("http://bioconductor.org/biocLite.R");biocLite("SNPRelate")

  
# LOAD R PACKAGES
library( SNPRelate )

# PREPARE INPUT DATA
genofile = openfn.gds(snpgdsExampleFileName())

# PRUNE SNPs BASED ON LINKAGE DISEQUILIBRIUM (LD)
snpset = snpgdsLDpruning(genofile, ld.threshold=0.2); snpset.id = unlist(snpset)

# PEFORM PCA
pca = snpgdsPCA(genofile, 
                maf=0.05, 
                missing.rate=0.05, 
                snp.id=snpset.id, 
                num.thread=2)

# DRAW A PLOT
plot(pca$eigenvect[,2], pca$eigenvect[,1], 
     xlab="Principal Component 2",
     ylab="Principal Component 1", 
     type="n")
race = as.factor(read.gdsn( index.gdsn(genofile, index=c("sample.annot", "pop.group"))))
table( race )
points(pca$eigenvect[,2], pca$eigenvect[,1], col=race)
legend("bottomleft", 
       title="Population", 
       legend=levels(race), 
       text.col=1:nlevels(race))
```
